include/cuco/detail/static_map/reduction_functors.cuh

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright (c) 2024, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+namespace cuco::reduce {
+
+/**
+ * @brief Device functor performing sum reduction, used with `insert-or-apply`
+ */
+struct plus {
```

Comment:
Since these are user-facing structs, we have to forward-declare them in public namespace (`detail` namespace is considered invisible to the user). Where should we put them? `operator.hpp` maybe?

Reply:
Since these functors will only be used with **static_map** API, we can just include this header in `static_map.cuh`. As these are not in detail namespace, user can directly use as `cuco::reduce::plus` .

Reply:
Right, they are not in `detail` namespace but the header is in the `cuco/detail` sub-directory.

These functors can be used from both `static_map` and `static_map_ref` so we would have to forward-decl them in both I guess.

Reply:
Does `cuco/utility/reduction_functors.cuh` make sense to you? I'm not saying this is better but just throwing ideas

Reply:
moved to `cuco/utility/reduction_functors.cuh` in https://github.com/NVIDIA/cuCollections/pull/555/commits/5ba022f023be54f40ad916c48571ddec63721034